lib/matplotlib/colorbar.py

```diff
@@ -674,9 +674,9 @@ def _process_values(self, b=None):
 
             b = self.norm.inverse(self._uniform_y(self.cmap.N + 1))
             if self._extend_lower():
-                b[0] = b[0] - 1
+                b[0] = 0.9 * b[0]
```

Comment:
I haven't completely thought this through, but it looks to me like using a multiplier is fine for the specific case where the problem arose--a log norm--but not for the case of a linear norm.  If `b[0]` or `b[1]` is zero, the multiplier doesn't do anything at all.
It also looks like the extend logic here might not even be needed, given the extend logic in `_uniform_y()`.  I haven't checked to see how much of that was added when the ends were modified to be of variable length.
The point of all this is just to find extended b values that will be normed to less than 0 if "under" or more than 1 if "over".